common/optimiser/Optimiser.java

```diff
@@ -189,7 +225,7 @@ static Status of(MPSolver.ResultStatus mpStatus) {
 
     @Override
     public String toString() {
-        return "Optimiser[" + "hasSolver=" + hasSolver + ", variables=" + variables.size() +
-                ", constraints=" + constraints.size() + "]" + (hasSolver ? solver.exportModelAsLpFormat() : "");
+        return "Optimiser[" + "hasSolver=" + (solver == null) + ", variables=" + variables.size() +
+                ", constraints=" + constraints.size() + "]" + (solver != null ? solver.exportModelAsLpFormat() : "");
```

Comment:
`hasSolver` ⇔ `solver != null`, which is correct in the second case, but not the first.